async-nats/src/jetstream/consumer.rs

```diff
@@ -15,6 +15,99 @@ use std::time::Duration;
 use serde::{Deserialize, Serialize};
 use time::serde::rfc3339;
 
+use crate::Error;
+
+pub trait IntoConsumerConfig {
+    fn into_consumer(self) -> ConsumerConfig;
```

Comment:
But probably returns Result<ConsumerConfig>, so we can validate?

Reply:
There is no need to validate this way, as we're going Pull/Push specific config -> config.